static/js/containers/CourseSearchPage.js

```diff
@@ -0,0 +1,322 @@
+    clearSearch()
+    if (!loaded && !facetChoiceProcessing) {
+      this.loadFacetChoices()
+    }
+    if (text) {
+      this.runSearch()
+    }
+  }
+
+  loadFacetChoices = async () => {
+    const { dispatch } = this.props
+    dispatch(actions.coursefacets.get())
+  }
+
+  loadMore = async () => {
+    const { searchLoaded } = this.props
+
+    if (!searchLoaded) {
+      // this function will be triggered repeatedly by <InfiniteScroll />, filter it to just once at a time
+      return
+    }
+
+    await this.runSearch({
+      incremental: true
+    })
+  }
+
+  runSearch = async (params: SearchInputs = { incremental: false }) => {
+    const {
+      clearSearch,
+      history,
+      location: { pathname, search },
+      runSearch
+    } = this.props
+
+    const text = params.text || this.state.text || undefined
+    const platforms = params.platforms || this.state.platforms || undefined
+    const topics = params.topics || this.state.topics || undefined
+    const availabilities =
+      params.availabilities || this.state.availabilities || undefined
+
+    const error = validateSearchQuery(text)
+    if (error) {
+      clearSearch()
+      this.setState({ error })
+      return
+    }
+
+    const type = SEARCH_FILTER_COURSE
+    history.replace({
+      pathname: pathname,
+      search:   qs.stringify({
+        ...qs.parse(search),
+        q: text,
+        type,
+        p: platforms,
+        t: topics,
+        a: availabilities
+      })
+    })
+    let from = this.state.from + SETTINGS.search_page_size
+    if (!params.incremental) {
+      clearSearch()
+      from = 0
+    }
+    this.setState({ from, error })
+    await runSearch({
+      channelName:    null,
+      text,
+      type,
+      from,
+      size:           SETTINGS.search_page_size,
+      platforms:      toArray(platforms),
+      topics:         toArray(topics),
+      availabilities: toArray(availabilities)
+    })
+  }
+
+  onUpdateFacets = async (e: Object) => {
+    const { text } = this.state
+    if (e.target.checked) {
+      await this.setState({
+        [e.target.name]: _.union(this.state[e.target.name], [e.target.value])
+      })
+    } else {
+      await this.setState({
+        [e.target.name]: _.without(this.state[e.target.name], e.target.value)
+      })
+    }
+    if (text) {
+      this.runSearch()
+    }
+  }
+
+  updateText = (event: ?Event) => {
+    // $FlowFixMe: event.target.value exists
+    const text = event ? event.target.value : ""
+    this.setState({ text })
+  }
+
+  renderResults = () => {
+    const { results, searchProcessing, initialLoad, total } = this.props
+    const { from } = this.state
+
+    if (searchProcessing && initialLoad) {
+      return <PostLoading />
+    }
+
+    if (!results.length) {
+      return (
+        <div className="empty-list-msg">There are no results to display.</div>
+      )
+    }
+
+    return (
+      <InfiniteScroll
+        // TODO: fix
```

Comment:
What does this mean?

Reply:
Whatever it was, I think it's fixed now :)

Reply:
Haha, can you remove the line here and in `SearchPage.js` as well?